### PR TITLE
Add displayMode: 'cards' to relationship field in new interfaces

### PR DESCRIPTION
**Pull request by mitchellhamilton**
cards display mode demo

### CODE REVIEW

File: packages-next/admin-ui/src/pages/ItemPage/index.tsx [hunk 1]
```diff
@@ -29,6 +29,11 @@ import { Fields } from './Fields';
 import { GraphQLError } from 'graphql';
 import { CreateItemDrawer } from '../../components/CreateItemDrawer';
 
+export { useInvalidFields, Fields };
+export * from '../../utils/getRootGraphQLFieldsFromFieldController';
+export * from '../../utils/serialization';
+export * from '../../utils/dataGetter';
```
Comment (mitchellhamilton) [inline packages-next/admin-ui/src/pages/ItemPage/index.tsx:35]: I really don't like this but but ehhhhh, it's probably fine

**Comment by gautamsi** [comment]
I hope this display mode can also be plugged in.

Also can the displaymode be changed in some value?

**Comment by mitchellhamilton** [comment]
> I hope this display mode can also be plugged in.

Not sure what you mean - you can replace the Field component to be whatever you want if that's what you're asking?

> Also can the displaymode be changed runtime with based on value in some other field?

It cannot, the general changing of fields based on other fields is something that @JedWatson has been wanting but it hasn't been figured out yet

**Comment by gautamsi** [comment]
By plugged in I meant if I could create my own field mode without replacing the full field view. But as replaying the field view is easy it should work.

**Comment by JedWatson** [comment]
> I could create my own field mode without replacing the full field view

Just a note on this, we're planning to make it so you can provide your own "card" (not the edit mode, the normal one) view by extending the custom views object. So it should be pretty flexible.

The create and edit views are the normal "create/edit" form UI for the set of fields you specify; custom fields and field views should _just work_ in this context as well.

**Comment by JedWatson** [comment]
Nice work @mitchellhamilton 

Some things left to do, from my basic testing:

- Add an existing item (should be an opt-in thing, with a config key like `linkExisting` or something. Displays a button next to the "create" button that reveals an autocomplete-like select, where you can find a new item to link like the "Select" field mode and then click "link {list.singular}")
- Focus the first field in the form when you click "edit"
- Reset changes (on the page item) doesn't reset changes to the relationships (e.g remove a related item, then click "reset changes" on the whole form, and the removed item doesn't reappear)

Separately, I noted a bug with our segmented-control selects; they visually select the first value, but when you save it wasn't actually selected. You have to interact with the control to set its value.

To reproduce, add a new phone number in this example and just enter a number then click "Create Phone Number"

**Comment by mitchellhamilton** [comment]
> Focus the first field in the form when you click "edit"

That already works, it's just that the segmented control doesn't accept `autoFocus`, just fixed up a number of other fields autoFocus though.

> Separately, I noted a bug with our segmented-control selects; they visually select the first value, but when you save it wasn't actually selected. You have to interact with the control to set its value.

I'm not gonna fix that in this PR

File: packages-next/admin-ui-utils/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+export * from './dataGetter';
```
Comment (mitchellhamilton) [inline packages-next/admin-ui-utils/src/index.ts:1]: Made this so there's a place to put utils that will likely be more unstable than the admin ui itself so this package can be major bumped without the admin ui being major bumped

**Review by JedWatson — APPROVED** [review]

**Pull request merged**